test/SignalProvider.test.js

```diff
@@ -27,6 +27,7 @@ describe('SignalProvider', () => {
     const c = shallow(<SignalProvider features={features}><div/></SignalProvider>);
     expect(c.find('div').length).toBe(0);
     c.instance().updateSignals({showStuff: true});
+    c.update();
```

Comment:
So this was the trick to handle the async behavior? Wow.

Reply:
Enzyme is allowing the object reference to get out of sync with the actual state of the object under test. This update method gets the props, dom structure, etc back in line with the current state of the originally referenced instance. I was under the impression that enzyme was going to be updated to allow us to get by without adding these but they are not. So to complete our upgrade we will need to add a lot of these...

https://github.com/airbnb/enzyme/issues/1175